lambdas/accesstoken/src/main/java/uk/gov/di/ipv/cri/address/api/handler/AccessTokenHandler.java

```diff
@@ -0,0 +1,115 @@
+package uk.gov.di.ipv.cri.address.api.handler;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.nimbusds.common.contenttype.ContentType;
+import com.nimbusds.oauth2.sdk.AccessTokenResponse;
+import com.nimbusds.oauth2.sdk.AuthorizationCodeGrant;
+import com.nimbusds.oauth2.sdk.ErrorObject;
+import com.nimbusds.oauth2.sdk.OAuth2Error;
+import com.nimbusds.oauth2.sdk.ParseException;
+import com.nimbusds.oauth2.sdk.TokenRequest;
+import com.nimbusds.oauth2.sdk.TokenResponse;
+import com.nimbusds.oauth2.sdk.http.HTTPRequest;
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.ipv.cri.address.library.helpers.ApiGatewayResponseGenerator;
+import uk.gov.di.ipv.cri.address.library.persistence.item.AddressSessionItem;
+import uk.gov.di.ipv.cri.address.library.service.AddressSessionService;
+import uk.gov.di.ipv.cri.address.library.validation.ValidationResult;
+
+import java.net.URI;
+
+public class AccessTokenHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private final AddressSessionService addressSessionService;
+
+    public AccessTokenHandler(AddressSessionService addressSessionService) {
+        this.addressSessionService = addressSessionService;
+    }
+
+    public AccessTokenHandler() {
+        this.addressSessionService = new AddressSessionService();
+    }
+
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        try {
+            TokenRequest tokenRequest = createTokenRequest(input.getBody());
```

Comment:
Could we move all validation into AddressSessionService, and have that method return a TokenRequest?